### PR TITLE
fix: locale format for cards

### PR DESCRIPTION
**Pull request by simson1**
Closes #

**Summary**

- Fix locale format

**Change List (commits, features, bugs, etc)**

- For all cards default date and tooltip format changed to 'L HH:mm:ss'.
- <DatePicker of the carbon library need locale in 2 digits for that we don't need to set locale of 2 digit in carbon-addons code.
- Ex. For locale 'en-gb' and 'en' date format is diffrent. so we should not set locale in 2 digits.

**Acceptance Test (how to verify the PR)**

- tests here

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

File: packages/react/src/components/DateTimePicker/DateTimePicker.story.jsx [hunk 1]
```diff
@@ -56,7 +56,6 @@ export const Default = () => {
     >
       <DateTimePicker
         id="datetimepicker"
-        dateTimeMask={text('dateTimeMask', 'YYYY-MM-DD HH:mm')}
```
Comment (herleraja) [inline packages/react/src/components/DateTimePicker/DateTimePicker.story.jsx:59]: any reason it was removed ?

Comment (simson1) [reply packages/react/src/components/DateTimePicker/DateTimePicker.story.jsx:59]: we already have default value in DateTimePicker.

Comment (herleraja) [reply packages/react/src/components/DateTimePicker/DateTimePicker.story.jsx:59]: This is for story book. In storybook they can change the value and test it out.

Comment (herleraja) [reply packages/react/src/components/DateTimePicker/DateTimePicker.story.jsx:59]: image

Comment (simson1) [reply packages/react/src/components/DateTimePicker/DateTimePicker.story.jsx:59]: reverted.

**Review by herleraja — APPROVED** [review]
LGTM

**Pull request merged**